### PR TITLE
merge develop to master

### PR DESCRIPTION
**Pull request by baonguyenhsts**

### CODE REVIEW

**Review by loc-nguyenthien — APPROVED** [review]

File: src/Chat/ChatProvider.tsx [hunk 1]
```diff
@@ -92,79 +84,63 @@ export const ChatProvider = React.forwardRef<any, ChatScreenProps>(
...
           setLoadEarlier(false);
-          getMoreMessage(conversationRef.current.id, enableEncrypt).then(
+
+          FirestoreServicesInstance.getMoreMessage().then(
             (data: MessageProps[]) => {
               setIsLoadingEarlier(false);
               setLoadEarlier(true);
-              if (data.length > 0) {
-                setMessagesList([...messagesList, ...data]);
+              if (data && data.length > 0) {
```
Comment (honghoangsts) [inline src/Chat/ChatProvider.tsx:92]: `data?.length`

File: src/Chat/ChatProvider.tsx [hunk 2]
```diff
@@ -92,79 +84,63 @@ export const ChatProvider = React.forwardRef<any, ChatScreenProps>(
...
-          conversationRef.current.id,
-          messages.text,
-          conversationRef.current.members,
-          file,
-          enableEncrypt
-        );
-      },
-      [enableEncrypt, userInfo.id, memberId]
-    );
+      if (messages?.type?.includes('image')) {
```
Comment (honghoangsts) [inline src/Chat/ChatProvider.tsx:115]: switch case?

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/Chat/ChatProvider.tsx (hunk 2 above)
Comment (baonguyenhsts) [reply src/Chat/ChatProvider.tsx]: updated

File: src/Chat/ChatProvider.tsx (hunk 1 above)
Comment (baonguyenhsts) [reply src/Chat/ChatProvider.tsx]: updated

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**